### ISSUE TITLE
Search and compare: Sometimes no changes are shown for files that do have changes

### ISSUE THREAD

**Issue by bluenote10** [gitkraken/vscode-gitlens#2661]
### Description

1. In "Search & Compare" I select "Compare References".
2. In the dropdown I select "Working Tree".
3. In the next dropdown I select e.g. `my_other_feature_branch`.
4. Now the "Search & Compare" view shows a subtree say "X files have changed". This list seems correct, i.e., in line with what `git diff my_other_feature_branch` lists.
5. When clicking on the files, the diff view opens, but it shows no changes at all (and pressing `ALT` + `F5` to cycle through changes has no effect). This seems to be wrong, because `git diff my_other_feature_branch` clearly shows changes between the two branches for the particular file. Both the left and the right side show the state that correspond to the HEAD/working directory, even though one of them should be different, because the file is different on `my_other_feature_branch` (note: The header of the tab does show a hash for the left-hand-side file, but none for the right-hand-side file, which seems okay). For some reason, this issue happens only for certain files. In some case the diff is correct.

### GitLens Version

13.5.0

### VS Code Version

Version: 1.77.3
Commit: 704ed70d4fd1c6bd6342c436f1ede30d1cff4710
Date: 2023-04-12T09:16:52.732Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Linux x64 5.4.0-146-generic
Sandboxed: No

### Git Version

git version 2.25.1

**Comment by eamodio** [comment]
@bluenote10 Logs might help some, but a video/Gif of what you are seeing would be helpful. Or a reproducible case -- as I use the "Compare Working Tree with <x>" feature regularly and haven't seen what you are experiencing.

**Comment by bluenote10** [comment]
Yes, the functionality also works for me most of the time. I was just trying to reproduce it in a publicly available repository, but all attempts on the repos/branches I tried worked so far. The particular repo were the issue occurs is a repository on which I'm not allowed to disclose any information. So it will be tricky to provide anything reproducible, but I'll see what I can do.

> Logs might help some, but a video/Gif of what you are seeing would be very helpful.

The video would literally just show a side-by-side diff view but without any visible diffs (after clicking on one of the affected files that are listed as files with changes) -- not so much insight from that except that it just doesn't work in this particular case.

**Comment by bluenote10** [comment]
Okay, I think I have found a minimal reproducible example.

First, let's setup a minimal repository with a branch named `main`:

```sh
mkdir test_repo
cd test_repo

git init
git checkout -b main

cat > some_file.txt <<EOL
Some file with example content.
The file contains two sections so that we can create a git merge
without a conflict.

Section A:
- 1
- 2
- 3

Section B:
- 1
- 2
- 3
EOL

git add .
git commit -m "initial commit"
```

Now, let's create a feature branch called `foo` that modifies section A in that file:

```sh
git checkout -b foo

cat > some_file.txt <<EOL
Some file with example content.
The file contains two sections so that we can create a git merge
without a conflict.

Section A:
- 1
- 2
- 3
- 4

Section B:
- 1
- 2
- 3
EOL

git add .
git commit -m "modified section A on branch foo"
```

Now, let's simulate a simultaneous change of that file but in section B on the `main` branch:

```sh
git checkout main

cat > some_file.txt <<EOL
Some file with example content.
The file contains two sections so that we can create a git merge
without a conflict.

Section A:
- 1
- 2
- 3

Section B:
- 1
- 2
- 3
- 4
EOL

git add .
git commit -m "modified section B on branch main"
```

Now let's merge the change on `main` back into the `foo` branch.

```sh
git checkout foo
git merge main
```

Note that on the `foo` branch now both sections are modified, because we have combined the two changes.

Now let's create a new feature branch called `bar` that branches off of `main`:

```sh
git checkout main
git checkout -b bar
```

From here you can use GitLens and do the "compare working set with" and select the `foo` branch. The "Search & Compare" tree view will show the file as modified, but when clicking on the file no change is visible. Both sides of the view show:

```
Some file with example content.
The file contains two sections so that we can create a git merge
without a conflict.

Section A:
- 1
- 2
- 3

Section B:
- 1
- 2
- 3
- 4
```

But that is wrong, because section A on branch `foo` has been modified (it contains a 4th element). Using `git diff foo` on branch `bar` properly shows the diff between `bar` and `foo`.

**Comment by bluenote10** [comment]
Any update on this? For me the "compare and search" feature is basically unusable, because GitLens constantly shows "no diff" where there actually is a diff. I'm wondering why nobody cares about that -- it seems such a fundamental bug 🤔

**Comment by gulench** [comment]
Ran into similar issues of late. Diff started showing after clearing the VSCode OUTPUT for GitLens.
